docs/read.md

```diff
@@ -382,14 +458,119 @@ _Can only be used with collections. Types can be a string, number, Date object,
 _Can only be used with collections. Types can be a string, number, Date object, or an array of these types, but not a Firestore Document Snapshot_
 
 ---
-## Advanced Usage
 
-## useCache
+## Firestore Indexes
+
+Firestore indexes are still required ([@see official Firestore Index documentation](https://firebase.google.com/docs/firestore/query-data/indexing)) for all complex queries of more than one property.
+
+Firestore queries are incredibly fast and do not degrade as the database gets
+ more and more documents. 
+
+There's a couple key points for Firestore reads:
+1. Each collection has a 40,000 index limit
+2. Each document has a 1MB limit
+3. Single indexes are automatically created.
+  In general this is good. But if you have a dynamic nested field in a document
+  (like `{orderItems: {'some-sku': 4.54, 'other-sku': 10.20}}`) ever key will 
+  get it's own index. [@see Firestore index exemptions](https://firebase.google.com/docs/firestore/query-data/index-overview)
+4. Complex queries are pre-defined. 
+  This means code can NOT dynamically mix and match different values to 
+  query data. The first time you query data that's not indexed the console will
+  notify you to create the index. 
+5. Queries are available offline.
+  Full queries work offline and in lo-fi/li-fi situations. The only limitation is 
+  queries will return only the documents that were previously downloaded. 
+
+
+---
+## Advanced Strategies
+
+## Rendering Lists
+
+How should we display a list of items? The ideal is that each
+React component will only rerender when the single document that it's
+displaying has changed. Also if the list of order of documents changes
+then only the parent component rerenders with the minimal changed. To seamlessly
+enable this
+
+```ts
+// useRead sets up a global subscription for changes & 
+//   will rerender when items are added/removed but NOT if a doc's values change
```

Comment:
I get a little confused here.

- If task name in task-294 changes, it won't be re-rendered 

- if a new task is included in Firestore, then it will re-render

correct?

Reply:
In this code if TASK-294 changes the `title` only the listItem for TASK-294 will be pre-rendered. This is because we use the list parent only gets an array id & path. That doesn't change so no render happens. Each item component also doesn't have any changes so only the component with the new title will pre-render.

Reply:
If a new task is add (or the order of the tasks is changed) then the parent will re-render. When it re-renders the key, id and path are the same for all the existing component so they will not re-render. Only the new item will be added and rendered in React.

Reply:
This is why the `useRead` and the `useCache` are so powerful. We can skip huge amount of React dirty checks and React renders because `useRead` and `useCache` can do a shallow equality check to skip React checks. When we only pass props to a component that are primitives then React is do a simple equality check and skip all expensive dirty checks and re-renders.

Reply:
@victorodgs is there a better way we can say this?

Reply:
Would it be better to say we should only render a list from IDs and then useCache at a lower level for display data like title?